File: src/main/scala/robosky/uplands/world/WorldRegistry.scala
package robosky.uplands.world

import net.fabricmc.fabric.api.dimension.v1.FabricDimensionType
import net.minecraft.world.World
import net.minecraft.world.dimension.DimensionType
import robosky.uplands.{UplandsMod, UplandsTeleporter}

object WorldRegistry {

  val UPLANDS_DIMENSION: FabricDimensionType = FabricDimensionType.builder()
    .factory((t: World, u: DimensionType) => new UplandsDimension(t, u)).skyLight(true)
    .defaultPlacer(UplandsTeleporter.ToUplandsBeacon)
    .buildAndRegister(UplandsMod :/ "luminiferous_uplands")

  def init(): Unit = Unit
}

File: src/main/java/robosky/uplands/world/WorldRegistry.java
package robosky.uplands.world;

import robosky.uplands.UplandsMod;
import robosky.uplands.mixin.client.SkyPropertiesAccessor;

import net.minecraft.util.registry.Registry;
import net.minecraft.util.registry.RegistryKey;
import net.minecraft.world.World;
import net.minecraft.world.dimension.DimensionType;

public final class WorldRegistry {

//    public static final FabricDimensionType UPLANDS_DIMENSION = FabricDimensionType.builder()
//        .factory(UplandsDimension::new)
//        .skyLight(true)
//        .defaultPlacer(UplandsTeleporter.TO_UPLANDS_BEACON)
//        .buildAndRegister(UplandsMod.id("luminiferous_uplands"));

    public static final RegistryKey<World> UPLANDS_WORLD_KEY = RegistryKey.of(Registry.DIMENSION, UplandsMod.id("luminiferous_uplands"));
    public static final RegistryKey<DimensionType> UPLANDS_DIMENSION_TYPE_KEY = RegistryKey.of(Registry.DIMENSION_TYPE_KEY, UplandsMod.id("luminiferous_uplands"));

    public static void init() {
        SkyPropertiesAccessor.getBY_IDENTIFIER().put(UplandsMod.id("uplands_sky"), new UplandsSkyProperties());
    }
}
